modules/packages/OrderedMap.chpl

```diff
@@ -0,0 +1,746 @@
+
+    if !isCopyableType(kt) || !isCopyableType(vt) then
+      compilerError("assigning orderedMap with non-copyable type");
+
+    lhs.clear();
+    for key in rhs.keys() {
+      lhs.add(key, rhs[key]);
+    }
+  }
+
+
+  /*
+    Returns `true` if the contents of two orderedMaps are the same.
+
+    :arg a: A orderedMap to compare.
+    :type a: orderedMap
+
+    :arg b: A orderedMap to compare.
+    :type b: orderedMap (with same keyType and valType)
+
+    :return: `true` if the contents of two orderedMaps are equal.
+    :rtype: `bool`
+  */
+  proc ==(const ref a: orderedMap(?kt, ?vt, ?ps), const ref b: orderedMap(kt, vt, ps)): bool {
+    if a.size != b.size then return false;
+    for (e1, e2) in zip(a.items(), b.items()) {
+      if e1 != e2 then return false;
+    } 
+    return true;
+  }
+
+  /*
+    Returns `true` if the contents of two orderedMaps are not the same.
+
+    :arg a: A orderedMap to compare.
+    :type a: orderedMap
+
+    :arg b: A orderedMap to compare.
+    :type b: orderedMap (with same keyType and valType)
+
+    :return: `true` if the contents of two orderedMaps are not equal.
+    :rtype: `bool`
+  */
+  proc !=(const ref a: orderedMap(?kt, ?vt, ?ps), const ref b: orderedMap(kt, vt, ps)): bool {
+    return !(a == b);
+  }
+
+  /*
+    Returns a new orderedMap containing the keys and values in either a or b.
+  */
+  proc +(a: orderedMap(?keyType, ?valueType, ?parSafe),
+         b: orderedMap(keyType, valueType, parSafe)) {
+    return a | b;
+  }
+
+  /*
+    Sets the left-hand side orderedMap to contain the keys and values in either
+    a or b.
+   */
+  proc +=(ref a: orderedMap(?keyType, ?valueType, ?parSafe),
+          b: orderedMap(keyType, valueType, parSafe)) {
+    a |= b;
+  }
+
+  /*
+    Returns a new orderedMap containing the keys and values in either a or b.
+  */
+  proc |(a: orderedMap(?keyType, ?valueType, ?parSafe),
+         b: orderedMap(keyType, valueType, parSafe)) {
+    var newMap = new orderedMap(keyType, valueType, parSafe, a.comparator);
+
+    for e in a.items() do newMap.add(e[0], e[1]);
+    for e in b.items() do newMap.add(e[0], e[1]);
+    return newMap;
+  }
+
+  /* Sets the left-hand side map to contain the keys and values in either
+     a or b.
+   */
+  proc |=(ref a: orderedMap(?keyType, ?valueType, ?parSafe),
+          b: orderedMap(keyType, valueType, parSafe)) {
+    // add keys/values from b to a if they weren't already in a
+    for e in b.items() do a.add(e[0], e[1]);
+  }
+
+  /*
+    Returns a new orderedMap containing the keys that are in both a and b.
+  */
+  proc &(a: orderedMap(?keyType, ?valueType, ?parSafe),
+         b: orderedMap(keyType, valueType, parSafe)) {
+    var newMap = new orderedMap(keyType, valueType, parSafe, a.comparator);
+    for (k, v) in a.items() {
+      if b.contains(k) then
+        newMap.add(k, v);
+    }
+    return newMap;
+  }
+
+  /* Sets the left-hand side orderedMap to contain the keys that are in both a and b.
+   */
```

Comment:
This could use a note about how it invalidates references to every element in `a`